nameserver/server.go

```diff
@@ -72,17 +147,14 @@ func rdnsHandler(config *dns.ClientConfig, lookups []Lookup) dns.HandlerFunc {
 	}
 }
 
-/* When we receive a request for a name outside of our '.weave.local.'
-   domain, ask the configured DNS server as a fallback.
-*/
-func notUsHandler(config *dns.ClientConfig) dns.HandlerFunc {
-	dnsClient := new(dns.Client)
-	dnsClient.UDPSize = UDPBufSize
+// When we receive a request for a name outside of our '.weave.local.'
+// domain, ask the configured DNS server as a fallback.
+func (s *DNSServer) notUsHandler(dnsClient *dns.Client) dns.HandlerFunc {
 	return func(w dns.ResponseWriter, r *dns.Msg) {
 		q := r.Question[0]
 		Debug.Printf("[dns msgid %d] Fallback query: %+v", r.MsgHdr.Id, q)
-		for _, server := range config.Servers {
-			reply, _, err := dnsClient.Exchange(r, fmt.Sprintf("%s:%s", server, config.Port))
+		for _, server := range s.config.Servers {
+			reply, _, err := dnsClient.Exchange(r, fmt.Sprintf("%s:%s", server, s.config.Port))
```

Comment:
Each _notUs_ DNS query results in an outbound DNS request, but we should have some sort of caching-resolver, an actor that is queried about names, mnages a cache, etc... so this simple solution that will be improved when zettio/weave#225 is fixed.